core/templates/domain/topic/topic-update.service.spec.ts

```diff
@@ -275,77 +275,149 @@ describe('Topic update service', function() {
     expect(UndoRedoService.getCommittableChangeList()).toEqual([]);
   });
 
-  it('should set/unset changes to a topic\'s name', function() {
-    expect(_sampleTopic.getName()).toEqual('Topic name');
-    TopicUpdateService.setTopicName(_sampleTopic, 'new name');
-    expect(_sampleTopic.getName()).toEqual('new name');
 
-    UndoRedoService.undoChange(_sampleTopic);
-    expect(_sampleTopic.getName()).toEqual('Topic name');
-  });
-
-  it('should create a proper backend change dict for changing names',
-    function() {
-      TopicUpdateService.setTopicName(_sampleTopic, 'new name');
-      expect(UndoRedoService.getCommittableChangeList()).toEqual([{
-        cmd: 'update_topic_property',
-        property_name: 'name',
-        new_value: 'new name',
-        old_value: 'Topic name'
-      }]);
-    }
-  );
+  /**
+   * Test coverage for:
+   *  setTopicName
+   *  setTopicDescription
+   *  setAbbreviatedTopicName
+   *  setTopicThumbnailFilename
+   *  setTopicThumbnailBgColor
+   *  setTopicLanguageCode
+   * on set/unset and backend change dict
+   */
+  [{
+    id: 'name',
+    getter: 'getName',
+    setter: 'setTopicName',
+    oldValue: 'Topic name'
+  },
+  {
+    id: 'description',
+    getter: 'getDescription',
+    setter: 'setTopicDescription',
+    oldValue: 'Topic description'
+  },
+  {
+    id: 'abbreviated_name',
+    getter: 'getAbbreviatedName',
+    setter: 'setAbbreviatedTopicName',
+    oldValue: undefined
+  },
+  {
+    id: 'thumbnail_filename',
+    getter: 'getThumbnailFilename',
+    setter: 'setTopicThumbnailFilename',
+    oldValue: undefined
+  },
+  {
+    id: 'thumbnail_bg_color',
+    getter: 'getThumbnailBgColor',
+    setter: 'setTopicThumbnailBgColor',
+    oldValue: undefined
+  },
+  {
+    id: 'language_code',
+    getter: 'getLanguageCode',
+    setter: 'setTopicLanguageCode',
+    oldValue: 'en',
+    overrideNewValue: 'fi'
+  }].forEach(function(test) {
+    const { id, getter, setter, oldValue, overrideNewValue } = test;
+
+    let newValue = overrideNewValue || 'new unique value';
+
+    it(`should set/unset changes to a topic's ${id}`, function() {
+      expect(_sampleTopic[getter]()).toEqual(oldValue);
+      TopicUpdateService[setter](_sampleTopic, newValue);
+      expect(_sampleTopic[getter]()).toEqual(newValue);
 
-  it('should set/unset changes to a topic\'s description', function() {
-    expect(_sampleTopic.getDescription()).toEqual('Topic description');
-    TopicUpdateService.setTopicDescription(_sampleTopic, 'new description');
-    expect(_sampleTopic.getDescription()).toEqual('new description');
+      UndoRedoService.undoChange(_sampleTopic);
+      expect(_sampleTopic[getter]()).toEqual(oldValue);
+    });
```

Comment:
I think this is not a good practice. Any thoughts? /cc @seanlip @nithusha21

Reply:
Here and elsewhere

Reply:
I agree. In general, don't use logic in tests; keep everything really simple to read. Cf. "DAMP" tests. 

Please fix, here and elsewhere.

Reply:
Will do. Thanks

Reply:
Done.